UC4-chaincode/src/main/java/de/upb/cs/uc4/chaincode/contract/admission/AdmissionContract.java

```diff
@@ -90,33 +98,72 @@ public String dropAdmission(final Context ctx, String admissionId) {
+     * @param enrollmentId enrollment to filter admissions by
+     * @param courseId     courseId to filter admissions by
+     * @param moduleId     moduleId to filter admissions by
      * @return Serialized List of Matching Admissions on success, serialized error on failure
      */
     @Transaction()
-    public String getAdmissions(final Context ctx, final String enrollmentId, final String courseId, final String moduleId) {
+    public String getCourseAdmissions(final Context ctx, final String enrollmentId, final String courseId, final String moduleId) {
         String transactionName = HyperledgerManager.getTransactionName(ctx.getStub());
+        final String[] args = new String[]{enrollmentId, courseId, moduleId};
+        ChaincodeStub stub = ctx.getStub();
+        try {
+            cUtil.validateApprovals(ctx, contractName,  transactionName, args);
+        } catch (SerializableError e) {
+            return e.getJsonError();
+        }
+        List<CourseAdmission> admissions = cUtil.getCourseAdmissions(stub, enrollmentId, courseId, moduleId);
+        try {
+            cUtil.finishOperation(stub, contractName,  transactionName, args);
+        } catch (SerializableError e) {
+            return e.getJsonError();
+        }
+        return GsonWrapper.toJson(admissions);
+    }
+
+    /**
+     * Gets ExamAdmissionList from the ledger.
+     *
+     * @param ctx          transaction context providing access to ChaincodeStub etc.
+     * @param admissionIds admissionIds to filter admissions by
+     * @param enrollmentId enrollment to filter admissions by
+     * @param examIds examIds to filter admissions by
+     * @return Serialized List of Matching Admissions on success, serialized error on failure
+     */
+    @Transaction()
+    public String getExamAdmissions(final Context ctx, final String admissionIds, final String enrollmentId, final String examIds) {
+        String transactionName = HyperledgerManager.getTransactionName(ctx.getStub());
+        final String[] args = new String[]{admissionIds, enrollmentId, examIds};
+        try {
+            cUtil.checkParamsGetExamAdmission(ctx, args);
+        } catch (SerializableError e) {
+            return e.getJsonError();
+        }
 
         ChaincodeStub stub = ctx.getStub();
         try {
-            cUtil.validateApprovals(ctx, contractName,  transactionName, new String[]{enrollmentId, courseId, moduleId});
+            cUtil.validateApprovals(ctx, contractName,  transactionName, args);
         } catch (SerializableError e) {
             return e.getJsonError();
         }
-        List<Admission> admissions = cUtil.getAdmissions(stub, enrollmentId, courseId, moduleId);
+        Type listType = new TypeToken<ArrayList<String>>() {}.getType();
+        ArrayList<String> admissionIdList = GsonWrapper.fromJson(admissionIds, listType);
+        ArrayList<String> examIdList = GsonWrapper.fromJson(examIds, listType);
```

Comment:
```suggestion
        List<String> admissionIdList = GsonWrapper.fromJson(admissionIds, listType);
        List<String> examIdList = GsonWrapper.fromJson(examIds, listType);
```
least restrictive type should be used